detect.py

```diff
@@ -164,16 +167,40 @@ def run(
                     if save_txt:  # Write to file
                         xywh = (xyxy2xywh(torch.tensor(xyxy).view(1, 4)) / gn).view(-1).tolist()  # normalized xywh
                         line = (cls, *xywh, conf) if save_conf else (cls, *xywh)  # label format
+
+                        folder_path = os.path.dirname(f'{txt_path}.txt')
+                        if not os.path.exists(folder_path):
+                            os.makedirs(folder_path)
                         with open(f'{txt_path}.txt', 'a') as f:
                             f.write(('%g ' * len(line)).rstrip() % line + '\n')
 
+                    if save_blurred_image:
+                        x1, y1 = int(xyxy[0].item()), int(xyxy[1].item())
+                        x2, y2 = int(xyxy[2].item()), int(xyxy[3].item())
+                        area_to_blur = im0[y1:y2, x1:x2]
+                        blurred = cv2.GaussianBlur(area_to_blur, (135, 135), 0)
+                        im0[y1:y2, x1:x2] = blurred
+
                     if save_img or save_crop or view_img:  # Add bbox to image
                         c = int(cls)  # integer class
                         label = None if hide_labels else (names[c] if hide_conf else f'{names[c]} {conf:.2f}')
                         annotator.box_label(xyxy, label, color=colors(c, True))
+
                     if save_crop:
                         save_one_box(xyxy, imc, file=save_dir / 'crops' / names[c] / f'{p.stem}.jpg', BGR=True)
 
+            # Save results (image with blurred detections)
+            if save_blurred_image:
+                folder_path = os.path.dirname(save_path)
+                if not os.path.exists(folder_path):
+                    os.makedirs(folder_path)
+
+                if not cv2.imwrite(
```

Comment:
is this an alternative to try catch?

Reply:
No, this is because if the image is not written is not throwing an error but it's just returning None.
So in this way we check if it worked or not.

Reply:
ahaa, alright!